File: output/mistralai_mistral_small_2603/few/python/Cálculo.py

import sys

def main():
    data = sys.stdin.read().split()
    if not data:
        return

    idx = 0
    M = int(data[idx])
    N = int(data[idx + 1])
    idx += 2

    X = list(map(int, data[idx:idx + M]))
    idx += M
    Y = list(map(int, data[idx:idx + N]))

    carry = 0
    result = []
    max_len = max(M, N)

    for i in range(max_len):
        bit_x = X[i] if i < M else 0
        bit_y = Y[i] if i < N else 0
        total = bit_x + bit_y + carry
        if total == 0:
            result.append(0)
            carry = 0
        elif total == 1:
            result.append(1)
            carry = 0
        elif total == 2:
            result.append(0)
            carry = 1
        else:  # total == 3
            result.append(1)
            carry = 1

    if carry:
        result.append(carry)

    print(' '.join(map(str, result)))

if __name__ == "__main__":
    main()

File: output/csbc_semish_2026/x_ai_grok_code_fast_1/zero/cpp/Cálculo.cpp

#include <bits/stdc++.h>
using namespace std;

int main() {
    int M, N;
    cin >> M >> N;
    vector<int> X(M);
    for(int i = 0; i < M; i++) cin >> X[i];
    vector<int> Y(N);
    for(int i = 0; i < N; i++) cin >> Y[i];
    int len = max(M, N);
    vector<int> result(len + 1, 0);
    int carry = 0;
    for(int i = len - 1; i >= 0; i--) {
        int bitX = (i < M) ? X[i] : 0;
        int bitY = (i < N) ? Y[i] : 0;
        int sum = bitX + bitY + carry;
        result[i + 1] = sum % 2;
        carry = sum / 2;
    }
    if(carry) result[0] = carry;
    int last = len;
    while(last >= 1 && result[last] == 0) last--;
    for(int i = 1; i <= last; i++) {
        cout << result[i];
        if(i < last) cout << " ";
        else cout << "\n";
    }
    return 0;
}
